athena/data/datasets/speech_synthesis.py

```diff
@@ -0,0 +1,258 @@
+                                  self.audio_featurizer.num_channels]),
+             "input_length": tf.TensorShape([1]),
+             "output_length": tf.TensorShape([1]),
+             "output": tf.TensorShape([None])}
+    """
+    default_config = {
+        "audio_config": {"type": "Fbank"},
+        "text_config": {"type":"vocab", "model":"athena/utils/vocabs/ch-en.vocab"},
+        "cmvn_file": None,
+        "remove_unk": True,
+        "input_length_range": [1, 10000],
+        "output_length_range": [20, 50000],
+        "data_csv": None
+    }
+
+    def __init__(self, config=None):
+        super().__init__()
+        # hparams
+        self.hparams = register_and_parse_hparams(
+            self.default_config, config, cls=self.__class__)
+        logging.info("hparams: {}".format(self.hparams))
+
+        self.audio_featurizer = AudioFeaturizer(self.hparams.audio_config)
+        self.feature_normalizer = FeatureNormalizer(self.hparams.cmvn_file)
+        self.text_featurizer = TextFeaturizer(self.hparams.text_config)
+        if self.hparams.data_csv is not None:
+            self.load_csv(self.hparams.data_csv)
+
+    def reload_config(self, config):
+        """ reload the config """
+        if config is not None:
+            self.hparams.override_from_dict(config)
+
+    def preprocess_data(self, file_path):
+        """ Generate a list of tuples (wav_filename, wav_length_ms, transcript speaker)."""
+        logging.info("Loading data from {}".format(file_path))
+        with open(file_path, "r", encoding="utf-8") as file:
+            lines = file.read().splitlines()
+        headers = lines[0]
+        lines = lines[1:]
+        lines = [line.split("\t") for line in lines]
+        self.entries = [tuple(line) for line in lines]
+
+        # handling speakers
+        self.speakers = []
+        if "speaker" not in headers.split("\t"):
+            entries = self.entries
+            self.entries = []
+            for wav_filename, wav_len, transcripts in entries:
+                self.entries.append(
+                    tuple([wav_filename, wav_len, transcripts, "global"])
+                )
+            self.speakers.append("global")
+        else:
+            for _, _, _, speaker in self.entries:
+                if speaker not in self.speakers:
+                    self.speakers.append(speaker)
+
+        # handling special case for text_featurizer
+        self.entries.sort(key=lambda item: len(item[2]))
+        if self.text_featurizer.model_type == "text":
+            _, _, all_transcripts, _ = zip(*self.entries)
+            self.text_featurizer.load_model(all_transcripts)
+
+        # apply some filter
+        self.filter_sample_by_unk()
+        self.filter_sample_by_input_length()
+        self.filter_sample_by_output_length()
+        return self
+
+    def load_csv(self, file_path):
+        """ load csv file """
+        return self.preprocess_data(file_path)
+
+    def __getitem__(self, index):
```

Comment:
I'm thinking we should probably add "speaker" in __getitem__ because lots of datasets requires speaker information. We could always set the value to zero if no speaker information is available.